packages/core/src/tracing/spanBuffer.js

```diff
@@ -483,3 +501,28 @@ function removeSpansIfNecessary() {
     spans = spans.slice(-maxBufferedSpans);
   }
 }
+/**
+ * @param {import('../core').InstanaBaseSpan} span
+ * @returns {import('../core').InstanaBaseSpan} span
+ */
+function processSpan(span) {
+  if (!span || typeof span.n !== 'string') {
+    return span;
+  }
+
+  // Currently, filter only if the ignoreEndpoint is configured.
+  // In the next phase, this check will be removed.
+  if (ignoreEndpoints) {
+    // @ts-ignore
+    span = applyFilter({ span, ignoreEndpoints });
+
+    if (!span) {
+      return null;
+    }
+  }
+  return transform(span);
+}
+// export the processSpan function for use in test.
+if (process.env.NODE_ENV === 'test') {
```

Comment:
issue: Could you please handle this differently? Very hard to maintain and not a good practise.

Reply:
Updated the code